app/task_routes.py

```diff
@@ -0,0 +1,131 @@
+from app import db
+from app.models.task import Task
+from flask import Blueprint, jsonify, abort, make_response, request
+import datetime
+import requests
+from app import token 
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+def validate_task(id):
+    try:
+        id = int(id)
+    except:
+        abort(make_response({"message": f"Task {id} is invalid"}, 400))
+
+    task = Task.query.get(id)
+
+    if not task:
+        abort(make_response({"message": f"Task {id} not found"}, 404))
+
+    return task
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    if request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body or "description" not in request_body:
+            return make_response(jsonify({"details": "Invalid data"}), 400)
+
+    new_task = Task(
+        title = request_body["title"],
+        description = request_body["description"],
+        completed_at = None
+    )
+    
+    db.session.add(new_task)
+    db.session.commit()
+    task_dict = dict(task=new_task.to_dict())
+    
+    return make_response(jsonify(task_dict), 201)
+    
+@tasks_bp.route("", methods=["GET"])
+def get_tasks():
+    sort = request.args.get("sort")
+    
+    if sort == "asc":
+        tasks = Task.query.order_by(Task.title.asc()).all()
+    else:
+        tasks = Task.query.order_by(Task.title.desc()).all()
```

Comment:
This scenario assumes that a user will only try to sort by "asc" and the then default would be to sort by "desc". 

This code could also be formatted so that there was an elif condition for the "desc" sorting and then an else block just to query without order_by.

```python
if sort == "asc":
       tasks = Task.query.order_by(Task.title.asc()).all()
elif sort == "desc":
       tasks = Task.query.order_by(Task.title.desc()).all()
else:
       tasks = Task.query.all()